### PR TITLE
Add ride ratings test

### PR DESCRIPTION
**Pull request by IntelOrca**
So we know if we have changed the calculation logic unintentionally.

### CODE REVIEW

File: test/tests/RideRatings.cpp [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+extern "C"
+{
+    #include <openrct2/platform/platform.h>
+    #include <openrct2/scenario/scenario.h>
+}
+
+class RideRatings : public testing::Test
+{
+protected:
+    static const char * ExpectedRideRatings[];
```
Comment (marijnvdwerf) [inline test/tests/RideRatings.cpp]: `char`

File: test/tests/RideRatings.cpp [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    void CalculateRatingsForAllRides()
+    {
+        for (int rideId = 0; rideId < MAX_RIDES; rideId++)
```
Comment (marijnvdwerf) [inline test/tests/RideRatings.cpp]: `int`

File: test/tests/RideRatings.cpp (hunk 1 above)
Comment (IntelOrca) [inline test/tests/RideRatings.cpp]: I assume you are suggesting I make it utf8? There is nothing other than ascii in this file.

**Comment by marijnvdwerf** [comment]
I'm not sure what' the ultimate idea with regard to the `(s|u)int(8|16|32)` types. There's #4936, but pretty much all new code seems to use normal ints etc.

**Comment by IntelOrca** [comment]
I should be using sint32, `int` is just a habbit.

**Comment by Gymnasiast** [comment]
What is the status of this?

**Comment by IntelOrca** [comment]
It needs an SV6 that has a reasonable amount of content in it, e.g. water, scenery the whole works with an instance of every single ride type in the park. Perhaps we can ask for one to be created on the forums?

File: src/openrct2/common.h [hunk 3]
```diff
@@ -294,6 +294,9 @@ typedef struct registers {
 assert_struct_size(registers, 7 * 4);
 #pragma pack(pop)
 
-#define UNUSED(x) ((void)(x))
+#define UNUSED(x)  ((void)(x))
+
+// Macro for turning symbol names into strings
+#define SZ(x)       #x
```
Comment (janisozaur) [inline src/openrct2/common.h]: What's wrong with [`nameof`](https://github.com/OpenRCT2/OpenRCT2/blob/eb0b29b/src/openrct2/common.h#L118)?

File: src/openrct2/core/Path.hpp [hunk 4]
```diff
@@ -23,6 +23,7 @@ namespace Path
 {
     utf8 * Append(utf8 * buffer, size_t bufferSize, const utf8 * src);
     std::string Combine(const std::string &a, const std::string &b);
+    std::string Combine(const std::string &a, const std::string &b, const std::string &c);
```
Comment (janisozaur) [inline src/openrct2/core/Path.hpp]: Why do we need two `Combine`s? Isn't it synonymous with `Combine(a, Combine(b, c))`?

File: src/openrct2/core/File.h [hunk 5]
```diff
@@ -27,4 +28,5 @@ namespace File
     bool Move(const std::string &srcPath, const std::string &dstPath);
     void * ReadAllBytes(const std::string &path, size_t * length);
     void WriteAllBytes(const std::string &path, const void * buffer, size_t length);
+    std::vector<std::string> ReadAllLines(const std::string &path);
```
Comment (janisozaur) [inline src/openrct2/core/File.h]: Why is it `File.h` and not `File.hpp`? I recall you provided a reason at some point, but I don't remember it.

File: test/tests/RideRatings.cpp [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+#include "TestData.h"
+
+extern "C"
+{
+    #include <openrct2/platform/platform.h>
+    #include <openrct2/game.h>
+}
+
+using namespace OpenRCT2;
+using namespace OpenRCT2::Audio;
```
Comment (janisozaur) [inline test/tests/RideRatings.cpp]: Audio?

File: src/openrct2/common.h (hunk 3 above)
Comment (IntelOrca) [inline src/openrct2/common.h]: Nothing, just didn't know we had it.

File: src/openrct2/core/Path.hpp (hunk 4 above)
Comment (IntelOrca) [inline src/openrct2/core/Path.hpp]: Why would you want to write `Combine(a, Combine(b, c))` over `Combine(a, b, c)`?

File: src/openrct2/core/File.h (hunk 5 above)
Comment (IntelOrca) [inline src/openrct2/core/File.h]: `hpp` contains implementation code.

File: test/tests/RideRatings.cpp (hunk 6 above)
Comment (IntelOrca) [inline test/tests/RideRatings.cpp]: Was originally needed, I can remove it now.

File: src/openrct2/core/Path.hpp (hunk 4 above)
Comment (janisozaur) [inline src/openrct2/core/Path.hpp]: For the same reason you would only have `int add(int a, int b)` and no `int add(int a, int b, int c)`, `int add(int a, int b, int c, int d)` and so on…

Comment (IntelOrca) [inline src/openrct2/core/Path.hpp]: I don't agree - the code clarity is better with an overload that takes three parameters. I would use vargs if possible.

Comment (janisozaur) [inline src/openrct2/core/Path.hpp]: Why not use variadic templates then? I'd rather see that than explicit implementation for n arguments.

Comment (IntelOrca) [inline src/openrct2/core/Path.hpp]: I don't know how they work. Can you provide an example or online example?

Comment (janisozaur) [inline src/openrct2/core/Path.hpp]: https://wandbox.org/permlink/Go1VaEWuAKJuzJuN

Comment (IntelOrca) [inline src/openrct2/core/Path.hpp]: thanks

File: test/tests/CMakeLists.txt [hunk 7]
```diff
@@ -117,8 +117,13 @@ target_link_libraries(test_string ${GTEST_LIBRARIES} test-common dl z)
...
 # Ride ratings test
-set(RIDE_RATINGS_TEST_SOURCES "${CMAKE_CURRENT_LIST_DIR}/RideRatings.cpp"
-                              "${CMAKE_CURRENT_LIST_DIR}/TestData.cpp")
-add_executable(test_ride_ratings ${RIDE_RATINGS_TEST_SOURCES})
-target_link_libraries(test_ride_ratings ${GTEST_LIBRARIES} libopenrct2 dl z)
-add_test(NAME ride_ratings COMMAND test_ride_ratings)
+if (NOT "z$ENV{CI}" STREQUAL "z")
+    message("Detected CI environment. Disabling ride rating test.")
+else ()
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink "${CMAKE_CURRENT_LIST_DIR}/testdata/parks" "${CMAKE_CURRENT_BINARY_DIR}/parks")
```
Comment (IntelOrca) [inline test/tests/CMakeLists.txt]: Can this be done at the top? Other tests may need testdata in the future.

Comment (IntelOrca) [inline test/tests/CMakeLists.txt]: This fails because you are symlinking parks rather than testdata.

Comment (janisozaur) [inline test/tests/CMakeLists.txt]: Yeah, but it still fails because I forgot I run all these tests in docker which has no `CI` env variable set…

**Pull request merged**